lib/dor/services/workflow_service.rb

```diff
@@ -7,539 +7,450 @@
+    # @return []
+    def create_workflow(repo, druid, workflow_name, wf_xml, opts = {:create_ds => true})
+      lane_id = opts.fetch(:lane_id, 'default')
+      ng_xml = wf_xml.is_a?(Nokogiri::XML::Document) ? wf_xml : Nokogiri::XML(wf_xml)
+      ng_xml.xpath('/workflow/process').each { |proc| proc['laneId'] = lane_id }
+      workflow_resource_method "#{repo}/objects/#{druid}/workflows/#{workflow_name}", 'put', ng_xml.to_s, {
+        :content_type => 'application/xml',
+        :params       => { 'create-ds' => opts[:create_ds] }
+      }
+    end
 
-      #
-      # Retrieves the raw XML for the given workflow
-      # @param [String] repo The repository the object resides in.  Currently recoginzes "dor" and "sdr".
-      # @param [String] druid The id of the object
-      # @param [String] workflow The name of the workflow
-      # @return [String] XML of the workflow
-      def get_workflow_xml(repo, druid, workflow)
-        workflow_resource_method "#{repo}/objects/#{druid}/workflows/#{workflow}"
-      end
+    # Updates the status of one step in a workflow.  Caller must handle any exceptions.
+    # Http Call
+    # ==
+    # The method does an HTTP PUT to the URL defined in `Dor::WF_URI`.  As an example:
+    #     PUT "/dor/objects/pid:123/workflows/GoogleScannedWF/convert"
+    #     <process name=\"convert\" status=\"completed\" />"
+    # @param [String] repo The repository the object resides in.  The service recoginzes "dor" and "sdr"
+    # @param [String] druid The id of the object
+    # @param [String] workflow The name of the workflow
+    # @param [String] process The name of the process step
+    # @param [String] status The status that you want to set -- using one of the values in @valid_statuses
+    # @param [Hash] opts optional values for the workflow step
+    # @option opts [Float] :elapsed The number of seconds it took to complete this step. Can have a decimal.  Is set to 0 if not passed in.
+    # @option opts [String] :lifecycle Bookeeping label for this particular workflow step.  Examples are: 'registered', 'shelved'
+    # @option opts [String] :note Any kind of string annotation that you want to attach to the workflow
+    # @option opts [String] :lane_id Id of processing lane used by the job manager.  Can convey priority or name of an applicaiton specific processing lane (e.g. 'high', 'critical', 'hydrus')
+    # @option opts [String] :current_status Setting this string tells the workflow service to compare the current status to this value.  If the current value does not match this value, the update is not performed
+    # @return []
+    def update_workflow_status(repo, druid, workflow, process, status, opts = {})
+      raise ArgumentError, "Unknown status value #{status}" unless @valid_statuses.include?(status.downcase)
+      opts = { :elapsed => 0, :lifecycle => nil, :note => nil }.merge!(opts)
+      opts[:elapsed] = opts[:elapsed].to_s
```

Comment:
`:elapsed` is init as integer.  The merge will take whatever the parameter provides.  Curious that it's cooerced to a string afterwards and not an integer (as in the init).